commons/src/main/java/fr/free/nrw/commons/upload/FilePathConverter.java

```diff
@@ -0,0 +1,49 @@
+package fr.free.nrw.commons.upload;
+
+
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.util.Log;
+
+public class FilePathConverter {
+
+    private Uri uri;
+    private Context context;
+
+    public FilePathConverter(Context context, Uri uri) {
```

Comment:
It looks like this is pretty much a one use object. Consider moving these parameters to `getFilePath(Context, Uri)`, making the method a static function, and converting the class to a utility (rename to BlahUtil, make a private constructor, and add final to the class).